### PR TITLE
Add hello-world OpenACC example

### PR DESCRIPTION
**Pull request by arporter**
Adds a single kernel, compilable (and executable!) example of using PSyclone to add OpenACC to an application. I've taken the opportunity to add a chapter on the examples to the documentation so that a user can find them more easily.

### CODE REVIEW

**Comment by arporter** [comment]
As a final check I tried compiling the example with PGI 18.10 on glados. This revealed some problems which I have now fixed. The resulting code compiles and runs on the GPU.
Ready for first review from @rupertford.

File: examples/gocean/eg1/README [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#------------------------------------------------------------------------------
+# Author A. R. Porter, STFC Daresbury Lab
+
+These scripts and this version of PSyclone work with version 1.0 of GOcean.
+
+In order to use PSyclone you must first install it, ideally with pip.
+See ../../README.md for more details.
```
Comment (rupertford) [inline examples/gocean/eg1/README]: Extra ../ here please.

File: examples/gocean/eg2/README [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+This is a very simple, single-kernel example of the use of PSyclone to
+generate an application that will run on a GPU. The kernel simply
+increments a field by adding the current value of the time-stepping
+loop index.
+
+The Makefile and this version of PSyclone work with version 1.0 of the GOcean
+API.
+
+In order to use PSyclone you must first install it, ideally with pip.
+See ../../README.md for more details.
```
Comment (rupertford) [inline examples/gocean/eg2/README]: Extra ../ please

File: examples/gocean/eg2/acc_transform.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+from __future__ import print_function
```
Comment (rupertford) [inline examples/gocean/eg2/acc_transform.py]: Copyright notice please.

File: examples/gocean/eg2/acc_transform.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    # Add an enter-data directive
+    newschedule, _ = dtrans.apply(schedule)
+
+    # Put an 'acc routine' directive inside each kernel
+    for kern in schedule.kern_calls():
+        _, _ = ktrans.apply(kern)
+        # Ideally we would module-inline the kernel here (to save having to
+        # rely on the compiler to do it) but this does not currently work
+        # for the fparser2 AST (issue #).
```
Comment (rupertford) [inline examples/gocean/eg2/acc_transform.py]: Which issue?

File: examples/gocean/eg2/alg.f90 [hunk 5]
```diff
@@ -0,0 +1,67 @@
+program alg
```
Comment (rupertford) [inline examples/gocean/eg2/alg.f90]: copyright notice please.

Comment (rupertford) [inline examples/gocean/eg2/alg.f90]: Copyright notice please.

File: examples/gocean/eg2/inc_field_mod.f90 [hunk 6]
```diff
@@ -0,0 +1,46 @@
+module inc_field_mod
```
Comment (rupertford) [inline examples/gocean/eg2/inc_field_mod.f90]: Copyright notice please.

File: examples/gocean/eg2/acc_transform.py (hunk 3 above)
Comment (rupertford) [inline examples/gocean/eg2/acc_transform.py]: Copyright norice please.

Comment (rupertford) [inline examples/gocean/eg2/acc_transform.py]: Copyright notice.

File: examples/gocean/eg2/acc_transform.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        #_, _ = itrans.apply(kern)
```
Comment (rupertford) [inline examples/gocean/eg2/acc_transform.py]: Warning here ...
acc_transform.py:43:9: E265 block comment should start with '# '

File: doc/examples.rst [hunk 8]
```diff
@@ -0,0 +1,109 @@
+.. _examples:
```
Comment (rupertford) [inline doc/examples.rst]: Copyright notice please.

File: doc/index.rst [hunk 9]
```diff
@@ -8,6 +8,7 @@
 
    introduction
    getting_going
+   examples
```
Comment (rupertford) [inline doc/index.rst]: Please remove commented out examples section further down in this file.

Comment (rupertford) [reply doc/index.rst]: Also, should we add a copyright notice and licence to this too.

File: doc/examples.rst (hunk 8 above)
Comment (rupertford) [reply doc/examples.rst]: Note, when I say copyright, I mean copyright and licence.

File: examples/gocean/eg2/acc_transform.py (hunk 4 above)
Comment (arporter) [reply examples/gocean/eg2/acc_transform.py]: Oop again. Completed.

File: examples/gocean/eg2/alg.f90 (hunk 5 above)
Comment (arporter) [reply examples/gocean/eg2/alg.f90]: Done.

File: examples/gocean/eg2/inc_field_mod.f90 (hunk 6 above)
Comment (arporter) [reply examples/gocean/eg2/inc_field_mod.f90]: Done.

File: examples/gocean/eg2/acc_transform.py (hunk 3 above)
Comment (arporter) [reply examples/gocean/eg2/acc_transform.py]: Done.

File: examples/gocean/eg2/acc_transform.py (hunk 7 above)
Comment (arporter) [reply examples/gocean/eg2/acc_transform.py]: Fixed.

File: doc/examples.rst (hunk 8 above)
Comment (arporter) [reply doc/examples.rst]: It's OK. I'm not taking you quite as literally today :-)

File: doc/index.rst (hunk 9 above)
Comment (arporter) [reply doc/index.rst]: Well spotted and done.

**Pull request merged**